7.1.js

```diff
@@ -0,0 +1,41 @@
+function SmartPlantEater() {
+    this.energy = 20;
+    this.dir = randomElement(directionNames);
+}
+
+SmartPlantEater.prototype.act = function(view) {
+    var space = view.find(" ");
+    if (this.energy > 120 && space) {

```

Comment:
nie rob prosze takich liczb nie wiadomo co znaczacych w kodzie.

to tak zwane magic numbers https://en.wikipedia.org/wiki/Magic_number_(programming)

sprawiaja, ze kod jest niejasny.

duzo lepiej przypisac opisowa nazwe do danej wartosci, np:
```javascript
var MINIMAL_ENERGY_FOR_REPRODUCTION = 120;
```

i pozniej uzywac tego w kodzie

stale rejestruj zawsze na samej gorze funkcji / elementu. pamietaj, ze w ES6 bedziesz uzywal do tego const zamiast var.